src/util.js

```diff
@@ -305,9 +307,49 @@ function getPrecedence(op) {
   return PRECEDENCE[op];
 }
 
+function getProjectOptions(searchDirectory) {
```

Comment:
This whole thing can be replaced by https://www.npmjs.com/package/read-pkg-up except for the part to keep searching if no prettier-entry